hooks/pre-commit

```diff
@@ -1,25 +1,26 @@
-#!/bin/bash
-if clang-format --version | grep -q 'version 11\.'; then
-   CLANG_FORMAT_EXECUTABLE="clang-format"
-else
-   CLANG_FORMAT_EXECUTABLE="clang-format-11"
-fi
+#!/bin/sh
+
+minVersion="14.0.0"
 
-if ! command -v $CLANG_FORMAT_EXECUTABLE &> /dev/null
-then
-    echo $CLANG_FORMAT_EXECUTABLE does not exist, make sure to install it
-    exit 1
+for file in $(find $(echo "$PATH" | tr ':' ' ') -maxdepth 1 -type f -executable -name 'git-clang-format*'); do
+  curName="$(basename "$file" | sed 's/^git-//')"
+  curVersion="$("$curName" --version | cut -d ' ' -f 3)"
+
+  if [ "$(printf '%s\n' "$curVersion" "$version" "$minVersion" | sort -V | tail -n 1)" = "$curVersion" ]; then
+    name="$curName"
+    version="$curVersion"
+  fi
+done
+
+if [ -z "$name" ]; then
+  echo "Could not find a suitable clang-format installation. Install clang-format that includes the git-clang-format script, with at least version $minVersion"
+  exit 1
 fi
 
-for FILE in $(git diff --cached --name-only)
-do
-	if [[ "$FILE" =~ src/[A-Za-z0-9\ \-]+*\.(c|h|cpp|cc)$ ]]; then
-		echo Autoformatting $FILE with $CLANG_FORMAT_EXECUTABLE
-		$CLANG_FORMAT_EXECUTABLE -style=file -i -- $FILE
-		git add -- $FILE
-	elif [[ "$FILE" =~ src/(components|displayapp|drivers|heartratetask|logging|systemtask)/.*\.(c|h|cpp|cc)$ ]]; then
-		echo Autoformatting $FILE with $CLANG_FORMAT_EXECUTABLE
-		$CLANG_FORMAT_EXECUTABLE -style=file -i -- $FILE
-		git add -- $FILE
-	fi
+changedFiles="$(git "$name" --extensions cpp,h --style file --staged -- ':!src/FreeRTOS' ':!src/libs')"
+
+echo "$changedFiles"
+
+echo "$changedFiles" | tail -n +2 | while read -r file; do
+  git add -- "$file"
```

Comment:
If I change multiple files and leave a formatting error in DisplayApp.cpp, the script will output the following, which is ambiguous and confusing. Do you think it needs to output anything?
```
changed files:
    src/displayapp/DisplayApp.cpp
```

Reply:
I guess this sort of applies to the error as well, since it might not be obvious it is coming from the pre-commit hook.

Reply:
Because someone has to enable the commit hook manually, I think that the error is fine as it is. What do you think?

The old pre-commit hook printed every file that it formatted quite explicitly. I personally don't think that's necessary, so I will remove that.